ofrak_core/ofrak/gui/server.py

```diff
@@ -1148,11 +1148,8 @@ async def set_projects_path(self, request: Request) -> Response:
     @exceptions_to_http(SerializedError)
     async def save_project_data(self, request: Request) -> Response:
         body = await request.json()
-        session_id = body["session_id"]
+        session_id = body["id"]
         project = self._get_project_by_id(session_id)
-        for binary_name, binary_metadata in body["binaries"].items():
-            project.binaries[binary_name].init_script = binary_metadata["init_script"]
-            project.binaries[binary_name].associated_scripts = binary_metadata["associated_scripts"]
```

Comment:
The frontend does make changes to the metadata itself, that's how the checkboxes work to set the associated & init scripts for each binary.

Reply:
> The frontend does make changes to the metadata itself, that's how the checkboxes work to set the associated & init scripts for each binary.

Ah I see... I think this is a bit confusing still then, I'm going to re-write the save function so that it takes the JSON from the request body and writes that to disc (minus ephemeral attributes such as `session_id` and `resource_ids`) so we don't have to select specific elements in the object that change in the frontend.